packages/utils/src/synergy.ts

```diff
@@ -0,0 +1,701 @@
+  // const conversationTask =
+  //   tasks.find((t) => t.name === "conversation-task") ||
+  //   (await client.ai.addTask("conversation-task", "llama", conversationPrompt, conversationExamples));
+  // console.log("tagTask", tagTask);
+  // // tag, topic change, & sungroup processing
+  // const tagResponse = await client.ai.prompt(
+  //   tagTask.taskId,
+  //   `{ existingTags: [${allTopics.map((t) => t.name).join(", ")}], newMessage: ${newItem.text} }`
+  // );
+  // console.log("tagResponse", tagResponse);
+  // const topicChangeResponse = await client.ai.prompt(
+  //   topicChangeTask.taskId,
+  //   `{ previousMessages: [${latestSubgroupItems.map((si) => si.text).join(", ")}], newMessage: ${newItem.text} }`
+  // );
+  // console.log("topicChangeResponse", topicChangeResponse);
+  // const subgroupResponse = await client.ai.prompt(
+  //   subgroupTask.taskId,
+  //   `{ previousMessages: [${latestSubgroupItems.map((si) => si.text).join(", ")}], newMessage: ${newItem.text} }`
+  // );
+  // console.log("subgroupResponse", subgroupResponse);
+  // const tags = JSON.parse(tagResponse);
+  // const changedSubject = JSON.parse(topicChangeResponse);
+  // const { newSubgroupName, newSubgroupSummary } = JSON.parse(subgroupResponse);
+  // // conversation processing
+  // const conversationResponse = await client.ai.prompt(
+  //   conversationTask.taskId,
+  //   `{ previousSubgroups: [${latestSubgroups.map((s) => s.summary).join(" <br/> ")}], newSubgroup: ${newSubgroupSummary} }`
+  // );
+  // console.log("conversationResponse", conversationResponse);
+  // let newConversationName,
+  //   newConversationSummary = "";
+  // try {
+  //   const parsed = JSON.parse(conversationResponse);
+  //   newConversationName = parsed.newConversationName;
+  //   newConversationSummary = parsed.newConversationSummary;
+  // } catch (e) {
+  //   console.error("Parsing error", e);
+  // }
+
+  // return {
+  //   topics: tags,
+  //   changedSubject,
+  //   newSubgroupName: newSubgroupName || "",
+  //   newSubgroupSummary: newSubgroupSummary || "",
+  //   newConversationName: newConversationName || "",
+  //   newConversationSummary: newConversationSummary || "",
+  // };
+}
+
+export function transformItem(type, item) {
+  // used to transform message, post, or task expressions into a common format
+  const newItem = {
+    type,
+    id: item.id,
+    author: item.author,
+    timestamp: item.timestamp,
+    text: "",
+    icon: "question",
+  };
+  if (type === "Message") {
+    newItem.text = item.body;
+    newItem.icon = "chat";
+  } else if (type === "Post") {
+    newItem.text = item.title || item.body;
+    newItem.icon = "postcard";
+  } else if (type === "Task") {
+    newItem.text = item.name;
+    newItem.icon = "kanban";
+  }
+  return newItem;
+}
+
+export async function findRelationships(perspective, itemId) {
+  return await new SubjectRepository(Relationship, {
+    perspective,
+    source: itemId,
+  }).getAllData();
```

Comment:
This is an interesting round-about way to do this :)
`SubjectEntity` has a static `all` function. I think we should add a `where({})` function too. There is `query({})` but that allows only for pagination, not constraining a property.

Something like this should be possible with what we have and without SubjectRepository:
```JS
let all = await Relationship.all()
return all.filter( r => r.expression == itemId)
```